Tests/GoTrueTests/GoTrueClientTests.swift

```diff
@@ -11,7 +11,15 @@ import ConcurrencyExtras
 
 @testable import GoTrue
 
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+
 final class GoTrueClientTests: XCTestCase {
+  fileprivate var api: APIClient!
+
+  #if !os(Linux)
+  // `fullfillment` doesn't exist on Linux yet.
```

Comment:
Please use [`wait(for:timeout:)`](https://developer.apple.com/documentation/xctest/xctestcase/2806856-wait) instead of fulfillment then.

Reply:
Looking at the sources again, I can't see where Linux doesn't actually have this, let me try to re-run this test and see what happens...

Reply:
So while I couldn't use `fulfillment` (doesn't exist on Linux), and couldn't use `wait` (since we're in an async context), I was able to use a simple semaphore to accomplish the same thing. Let's see how that does across platforms.

Reply:
Looks like that works well!